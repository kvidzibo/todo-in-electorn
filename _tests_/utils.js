import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

export const render = (comp, mode = 3) => {
  switch (mode) {
    case 1:
      return shallow(comp)
    case 2:
      return renderer.create(comp).toJSON()
    case 3:
      return {
        component: shallow(comp),
        tree: renderer.create(comp).toJSON()
      }
    case 4:
      return mount(comp)
    case 5:
      return {
        component: mount(comp),
        tree: renderer.create(comp).toJSON()
      }
  }
}
