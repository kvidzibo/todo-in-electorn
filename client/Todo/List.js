import React, { PureComponent } from 'react'
import { Field, Button } from '../Components'
import { handleChange } from '../Utils'
import { withRefetch } from '../HOC'
import Task from './Module'
import Item from './Item'

class List extends PureComponent {
  handleChange = handleChange.bind(this)
  state = {
    task: '',
    error: ''
  }
  add = () => {
    const { task } = this.state
    this.setState({error: '', task: ''})
    Task.save({task}).catch(e => {
      this.setState({error: e.message})
    })
  }

  render () {
    const { task, error } = this.state
    const { data } = this.props
    return (
      <div>
        <Field label="task" error={error}>
          <input value={task} onChange={this.handleChange} className="input" name="task" type="text"/>
        </Field>
        <Field>
          <Button onClick={this.add}>Add</Button>
        </Field>
        {!!data && data.map(item => (
          <Item key={item._id} item={item}/>
        ))}
      </div>
    )
  }
}

export default withRefetch(List, Task, () => Task.find({status: 'todo'}))
