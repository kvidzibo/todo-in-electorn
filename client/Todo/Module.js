import { remote } from 'electron'
const { Task } = remote.require('./app/modules')

export default Task
