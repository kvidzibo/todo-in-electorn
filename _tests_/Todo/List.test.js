import React from 'react'
import { render } from '../utils'
import List from '../../client/Todo/List'
import Module from '../../client/Todo/Module'

jest.mock('electron', () => ({
  remote: {require: jest.fn(() => ({Task: {}}))}
}))
jest.mock('../../client/Todo/Module')
jest.mock('../../client/HOC', () => ({
  withRefetch: (c) => c
}))

beforeEach(() => {
  Module.save = jest.fn().mockResolvedValue('')
})

test('renders correctly', () => {
  const { component, tree } = render(<List />)
  expect(component.find('Field').length).toEqual(2)
  expect(tree).toMatchSnapshot()
})

test('calls save on add', () => {
  const component = render(<List />, 1)
  component.find('Button').simulate('click')
  expect(Module.save.mock.calls.length).toBe(1)
})

test('sets error on save error', (done) => {
  Module.save = jest.fn().mockRejectedValue({message: 'error'})
  const { component, tree } = render(<List />)
  component.find('Button').simulate('click')
  expect(Module.save.mock.calls.length).toBe(1)
  setTimeout(() => {
    expect(component.state().error).toBe('error')
    expect(tree).toMatchSnapshot()
    done()
  }, 0)
})
