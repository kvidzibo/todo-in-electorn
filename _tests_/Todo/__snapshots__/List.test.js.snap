// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      task
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="task"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`sets error on save error 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      task
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="task"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
